packages/app/src/scripts/food.js

```diff
@@ -0,0 +1,214 @@
+let rotation = 0
+
+export const foodScript = (sketch) => {
+
+  let width = 400
+  let height = 400
+
+  let models = new Map()
+
+  let materials
+  let bgColor
+
+  /**
+   * p5: Setup the WebGL canvas that we will be
+   * drawing our objects to. Materials will also be
+   * initialized here for drawing later on.
+   * 
+   * This code is only ran when the canvas/applet
+   * is first rendered.
+   */
+  sketch.setup = () => {
+
+    sketch.createCanvas(width, height, sketch.WEBGL)
+    sketch.angleMode(sketch.DEGREES)
+    bgColor = sketch.color(199, 151, 197)
+
+    /**
+     * Load the different parts of the noodle bowl to
+     * be rendered. 
+     * 
+     * Q: Why are the parts separate files and not in
+     * a singular file? 
+     * 
+     * A: Normally, the surface material information of a 
+     * .obj file is included in a separate .mtl file
+     * that should be loaded alongside the main file.
+     * However, p5 does not parse material information,
+     * thus only allowing one material per one object
+     * file loaded. By loading the different parts of our
+     * model separately, we can apply different materials
+     * to each of them. This clearly is impractical for
+     * larger, more detailed models, but for our use case
+     * writing a custom .obj parser would be a colossal
+     * waste of development time.
+     */
+
+    customLoadModel('bowl', bowl)
+    customLoadModel('bowlRim', bowlRim)
+    customLoadModel('soup', soup)
+    customLoadModel('salad', salad)
+    customLoadModel('noodles', noodles)
+    customLoadModel('chopsticks', chopsticks)
+    customLoadModel('eggWhite', eggWhite)
+    customLoadModel('eggYolk', eggYolk)
+    customLoadModel('tofu', tofu)
+
+    materials = {
+      white: () => materialFunc(242, 242, 242, true),
+      blue: () => materialFunc(117, 198, 230, false),
+      soup: () => materialFunc(230, 173, 117, false),
+      salad: () => materialFunc(152, 255, 74, true),
+      noodle: () => materialFunc(255, 234, 176, false),
+      wood: () => materialFunc(128, 73, 73, false),
+      eggYolk: () => materialFunc(252, 186, 3, false)
+    }
+  }
+
+  /**
+   * p5: Draw the actual scene
+   */
+  sketch.draw = () => {
+
+    /**
+     * Destroy the p5 instance if not on the front page.
+     * This is to prevent multiple canvases from being
+     * initialized, when switching from the front page
+     * to, for example, the '/add' page, then switching
+     * back to the front page.
+     */
+    if (sketch.getURLPath().length !== 0) {
+      sketch.remove()
+    }
+
+    /* General rendering setup */
+
+    sketch.smooth()
+    sketch.noStroke()
+    sketch.background(bgColor)
+
+    /* Ambient lighting based on the background color helps tie the models to the scene */
+
+    let ambientStrength = 0.9
+    sketch.ambientLight(
+      sketch.red(bgColor) * ambientStrength,
+      sketch.green(bgColor) * ambientStrength,
+      sketch.blue(bgColor) * ambientStrength
+    )
```

Comment:
This doesn't do anything? The material functions override the ambient light anyway?

Reply:
This indeed *does* do things. Curious.